### PR TITLE
Add support for cl_ext_float_atomics

### PR DESCRIPTION
**Pull request by haonanya**
See https://github.com/KhronosGroup/OpenCL-Docs/blob/1a13ed588ddba99dcddfd345a008e0d5131a6501/extensions/cl_ext_float_atomics.asciidoc for details
Signed-off-by: haonanya <haonan.yang@intel.com>

### CODE REVIEW

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
++                                        double operand, memory_order order,
++                                        memory_scope scope);
++double __ovld atomic_fetch_max_explicit(volatile atomic_double *object,
++                                        double operand, memory_order order,
++                                        memory_scope scope);
++#endif
++
++#endif // cl_ext_float_atomics
++
++#if defined(cl_ext_float_atomics)
```
Comment (hewj03) [inline patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:156]: Can line 154 and line 156 be removed? They are the same guard cl_ext_float_atomics

Comment (haonanya) [reply patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch]: done

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
++#error "Incorrectly defined __opencl_c_ext_fp32_global_atomic_min_max"
++#endif
++#if __opencl_c_ext_fp16_local_atomic_min_max != 1
++#error "Incorrectly defined __opencl_c_ext_fp16_local_atomic_min_max"
++#endif
++#if __opencl_c_ext_fp32_local_atomic_min_max != 1
++#error "Incorrectly defined __opencl_c_ext_fp32_local_atomic_min_max"
++#endif
++
++#else
```
Comment (hewj03) [inline patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:300]: suggest to add comment after #else, "// ..."
and add an empty line after the line.

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
++#ifdef __opencl_c_ext_fp16_global_atomic_load_store
++#error "Incorrectly __opencl_c_ext_fp16_global_atomic_load_store defined"
++#endif
```
Comment (hewj03) [inline patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:303]: is there fp32_global_atomic_load_store support?

Comment (haonanya) [reply patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:303]: No, there are only __opencl_c_ext_fp16_global_atomic_load_store and __opencl_c_ext_fp16_local_atomic_load_store.
https://github.com/KhronosGroup/OpenCL-Docs/blob/1a13ed588ddba99dcddfd345a008e0d5131a6501/extensions/cl_ext_float_atomics.asciidoc

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch (hunk 2 above)
Comment (haonanya) [reply patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:300]: There is comment already on `#endif //(defined(__OPENCL_CPP_VERSION__) || __OPENCL_C_VERSION__ >= 200)`. Do we still need to add comment on `#else`? 
Thanks very much.

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch (hunk 3 above)
Comment (hewj03) [reply patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:303]: I see other #else in opencl-c.h doesn't have comment. So It should be fine.

File: patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch (hunk 2 above)
Comment (hewj03) [reply patches/clang/0003-OpenCL-support-cl_ext_float_atomics.patch:300]: I see other #else in opencl-c.h doesn't have comment. So It should be fine.

**Review by hewj03 — APPROVED** [review]
LGTM

**Pull request merged**